tests/kafkatest/tests/streams/streams_cooperative_rebalance_upgrade_test.py

```diff
@@ -45,7 +45,7 @@ class StreamsCooperativeRebalanceUpgradeTest(Test):
 
     streams_eager_rebalance_upgrade_versions = [str(LATEST_0_10_0), str(LATEST_0_10_1), str(LATEST_0_10_2), str(LATEST_0_11_0),
                                                 str(LATEST_1_0), str(LATEST_1_1), str(LATEST_2_0), str(LATEST_2_1), str(LATEST_2_2),
-                                                str(LATEST_2_3)]
+                                                str(LATEST_2_3), str(LATEST_2_4)]
```

Comment:
Here a would like to have confirmation: Does it make sense to test the rebalance upgrade from 2.4?
 
A system test failed (http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2019-12-17--001.1576588759--cadonna--update-systests-24--76c43d3/report.html) with the message `org.apache.kafka.common.config.ConfigException: Invalid value 2.4 for configuration upgrade.from: String must be one of: null, 0.10.0, 0.10.1, 0.10.2, 0.11.0, 1.0, 1.1, 2.0, 2.1, 2.2, 2.3`.

Reply:
No, can we maybe leave a comment here to clarify that this list is "fixed" and we should not add new versions to it? Since COOPERATIVE is on by default starting in 2.4, the 2.4 -> 2.5 upgrade path doesn't involve any changing rebalance protocols.

That said, it probably _will_ involve version probing, but the version probing system test doesn't actually test this...

Reply:
Just to clarify, we shouldn't add a new `UPGRADE_FROM` config either

Reply:
Not sure why we would need an `UPGRADE_FROM_24` config? We don't have any breaking changings in `2.5` (at least no yet) that would require this config? Not sure if we need this to upgrade form from `2.4` eager to `2.5` incremental -- I don't think we need it? Currently, if we remove the config, `2.4` will switch to incremental. If people would want to stay on eager, they would leave the `UPGRADE_FROM_23` config, even if they upgrade from 2.4. to 2.5.

\cc @ableegoldman

Reply:
Seem I did not refresh my browsers... Just seeing Sophie's reply now...

Reply:
Github's still on Web 1.0 😛

Reply:
> Just to clarify, we shouldn't add a new UPGRADE_FROM config either

> Not sure why we would need an UPGRADE_FROM_24 config? 

This is a misunderstanding on my side. So, as far as I understand, we just need `upgrade.from` for incompatible versions that need a specific upgrade path, right? I thought we always need to set `upgrade.from` for each upgrade.

Reply:
Correct.